tembo-pgmq-python/tembo_pgmq_python/queue.py

```diff
@@ -47,99 +48,101 @@ def __post_init__(self) -> None:
         with self.pool.connection() as conn:
             conn.execute("create extension if not exists pgmq cascade;")
 
+    @classmethod
+    def from_env(cls, env_file=".env"):
```

Comment:
I'm not a fan of this function.

I don't think libraries should load environment files automatically, but **at most** read environment variables, and even that is questionable IMO.

Reply:
I'm aiming to make the library as easy to use as possible. I added this function because I encountered some issues with pydantic-settings. If you're not satisfied with it, we can remove it. My goal was to improve testing

Reply:
I really like the testing changes, but this part strikes me as "not right". I'm not a python expert, though, so let's wait and see what others think :)

Reply:
I really like how this adds the ability to read the values from environment variables. I think I would prefer to have it skip the direct call to python-dotenv (let the user do that). Instead, if user provides values directly through arguments, use them, otherwise fall back to the environment variables (the names you chose are good). If the values weren't provided in either env var or arguments, then it should fail.

Reply:
so what do you think I should change?

Reply:
I think this would be very nice:

1. remove `from_env()` and the dependency on python-dotenv
2. expand on `__post_init__` to pull in values from environment variables (similar to how you have it in this PR) 
3. if any of the attributes are provided as arguments, override the env variables with the argument values
4. if we dont have enough information to produce a valid connection string, either fall back to the default values (localhost:5432/postgres) or raise an exception I would be ok going either way with this (use defaults, or require user to provide connection info). @v0idpwn any thoughts on this?

Reply:
Agreed!

Reply:
done! but I learn a lot, painfully. :)))))))